app/src/main/kotlin/at/bitfire/davdroid/ui/AccountsModel.kt

```diff
@@ -92,7 +96,33 @@ class AccountsModel @AssistedInject constructor(
     private val workManager = WorkManager.getInstance(context)
     private val runningWorkers = workManager.getWorkInfosFlow(WorkQuery.fromStates(WorkInfo.State.ENQUEUED, WorkInfo.State.RUNNING))
 
-    val accountInfos: Flow<List<AccountInfo>> = combine(accounts, runningWorkers) { accounts, workInfos ->
+    @OptIn(ExperimentalCoroutinesApi::class)
+    private val accountsPendingInSyncFramework: Flow<List<Account>> =
+        accounts.flatMapLatest { accounts ->
+            if (accounts.isEmpty())
+                flowOf(emptyList())
+            else {
+                // For each account, create a Flow<Boolean> emitting the state on whether it has a pending sync.
+                val pendingSyncCheckFlows: List<Flow<Account?>> = accounts.map { account ->
+                    syncFrameWork.isSyncPending(account, SyncDataType.entries)
+                        .map { hasPendingSync ->
+                            // Map this boolean answer to its Account if it is pending, or null if not.
+                            if (hasPendingSync) account else null
+                        }
+                }
+                // Combine all account flows Flow<Account?> into a single flow emitting a list of accounts with
+                // pending sync. The null values which we filter out are the non-pending accounts.
+                // Now, whenever any account's pending state changes, the combined flow emits the updated list.
+                combine(pendingSyncCheckFlows) { it.filterNotNull() }
```

Comment:
However I had not understood this because I was used to the two-argument syntax of `combine`. With that syntax it wouldn't be possible to combine the values just with a single call.

But it seems correct. I suggest to add this comment:

```kotlin
combine(pendingSyncCheckFlows) { combinedAccounts ->
    // combinedAccounts is an Array of the most recently emitted values of the pendingSyncCheckFlows,
    // i.e. an Array<Account?> – one entry for every pendingSyncCheckFlow that is either the account
    // name (sync pending) or null (no sync pending).
    combinedAccounts.filterNotNull()
}
```